test/models/events_test.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class EventTest < ActiveSupport::TestCase
+  setup do
+    @calendar = create(:calendar)
+  end
+
+  test 'ensure validation fails for duplicate event model' do
+    assert_raises ActiveRecord::RecordInvalid do
+      create_list(:event, 2,
+                  summary: 'Mom and Pops Pet Store',
+                  dtstart: DateTime.now - 1.hour,
+                  calendar: @calendar)
+    end
+  end
+
+  test 'can create two events with the same start date and calendar but diff summary' do
+    event_hash = { dtstart: DateTime.now - 1.hour, calendar: @calendar }
+    a = create(:event, summary: 'cycling triathalon', **event_hash)
+    b = create(:event, summary: 'foonlys get together :,)', **event_hash)
+    assert a
```

Comment:
ohh uhhh hmmm. i am not sure i like that you are using factories to test model validation code. it works but it seems odd- like why can't you just `Event.new` put in a bunch of data and then check `assert event_a.valid?`. in this way intent is explicitly defined in code. `assert a` is a very C flavour

Reply:
I mean I can, but that would require doing a bunch of stuff to either skip or satisfy location validation. The whole point of factories is allowing us to create models that satisfy X constraints and allow me to focus on testing, solely, 3 variations of ±the same object, without having to put all the work into constructing them.